tests/unit/parallel/algorithms/foreach_executors.cpp

```diff
@@ -0,0 +1,90 @@
+//  Copyright (c) 2015 Daniel Bourgeois
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/hpx_init.hpp>
+#include <hpx/hpx.hpp>
+
+#include "foreach_tests.hpp"
+
+////////////////////////////////////////////////////////////////////////////////
+template <typename ExPolicy>
+void test_executors(ExPolicy && policy)
+{
+    using iterator_tag = std::random_access_iterator_tag;
```

Comment:
Please use `typedef std::random_access_iterator_tag iterator_tag;` instead as the using syntax is not supported by gcc 4.6.